### PR TITLE
guake: 3.6.3 -> 3.9.0

### PR DESCRIPTION
**Pull request by yorickvP**
Fixes CVE-2021-23556

#166650

###### Description of changes

[Changelog](https://github.com/Guake/guake/releases)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siph** [comment]
Result of `nixpkgs-review pr 195210` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>guake</li>
  </ul>
</details>

**Review by siph — APPROVED** [review]
This builds and runs in KDE Plasma.

File: pkgs/applications/terminal-emulators/guake/default.nix [hunk 1]
```diff
@@ -55,12 +55,15 @@ python3.pkgs.buildPythonApplication rec {
     pycairo
     pygobject3
     setuptools
```
Comment (jtojnar) [inline pkgs/applications/terminal-emulators/guake/default.nix:57]: Pbr and setuptools can probably be removed.

**Comment by viraptor** [comment]
Testing this on KDE the app starts, but doesn't seem to register the hotkey. Manually running `guake-toggle` works.
Errors on startup:
```
Loading Gnome schema from: /nix/store/2hc7y4l5h3nk0hy23x7mzi5cblnwz5wz-guake-3.9.0/share/gsettings-schemas/guake-3.9.0/glib-2.0/schemas/
Schema from old guake version detected, regenerating schema
(...)
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.ServiceUnknown: The name is not activatable
Compiling schema: /nix/store/2hc7y4l5h3nk0hy23x7mzi5cblnwz5wz-guake-3.9.0/share/gsettings-schemas/guake-3.9.0/glib-2.0/schemas/
(...)
Failed to create file “/nix/store/2hc7y4l5h3nk0hy23x7mzi5cblnwz5wz-guake-3.9.0/share/gsettings-schemas/guake-3.9.0/glib-2.0/schemas/gschemas.compiled.OVBAU1”: Permission denied
(...)
Traceback (most recent call last):
  File "/nix/store/2hc7y4l5h3nk0hy23x7mzi5cblnwz5wz-guake-3.9.0/lib/python3.10/site-packages/guake/guake_app.py", line 125, in __init__
    try_to_compile_glib_schemas()
  File "/nix/store/2hc7y4l5h3nk0hy23x7mzi5cblnwz5wz-guake-3.9.0/lib/python3.10/site-packages/guake/paths.py", line 96, in try_to_compile_glib_schemas
    subprocess.check_call(["glib-compile-schemas", "--strict", SCHEMA_DIR])
  File "/nix/store/9srs642k875z3qdk8glapjycncf2pa51-python3-3.10.7/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['glib-compile-schemas', '--strict', '/nix/store/2hc7y4l5h3nk0hy23x7mzi5cblnwz5wz-guake-3.9.0/share/gsettings-schemas/guake-3.9.0/glib-2.0/schemas/']' returned non-zero exit status 1.
Loading Gnome schema from: /nix/store/2hc7y4l5h3nk0hy23x7mzi5cblnwz5wz-guake-3.9.0/share/gsettings-schemas/guake-3.9.0/glib-2.0/schemas/
```

Seems like something we should be running as part of the build, right?

**Comment by jtojnar** [comment]
We do.

That looks like a bug in Guake – on first launch the `schema-version` setting will be an empty string:

https://github.com/Guake/guake/blob/50c19446b5e6ffde10f0d4f849259d723054b9bd/guake/data/org.guake.gschema.xml#L10

but Guake expects it to match the Guake version, or it will attempt to recompile the schema:

https://github.com/Guake/guake/blob/50c19446b5e6ffde10f0d4f849259d723054b9bd/guake/guake_app.py#L121

They should probably modify the default value in the `gschemas.xml` at build time and then compare the [default value](https://docs.gtk.org/gio/method.SettingsSchemaKey.get_default_value.html) at runtime.

**Comment by viraptor** [comment]
If I understand it correctly, the schema will be loaded anyway after the failure. Could someone confirm the hotkey issue is just on my side and not a general problem to solve?

**Comment by yorickvP** [comment]
The hotkey works for me on xwayland apps (which is as good as this was gonna get on wayland).

**Comment by siph** [comment]
The toggle hotkey is working for me in KDE/X11 with no problems.

**Comment by viraptor** [comment]
Thank you.

**Pull request merged**